docs/style_guide.adoc

```diff
@@ -96,79 +94,24 @@ g_environment: someval
 ----
 
 === Ansible Role Variables
-Ansible role variables are defined as variables contained in (or passed into) a role.
-
 '''
-[[Role-variables-MUST-have-a-prefix-of-atleast-3-characters-See.below.for.specific.naming.rules]]
+[[Ansible-role-variable-prefix]]
 [cols="2v,v"]
 |===
-| <<Role-variables-MUST-have-a-prefix-of-atleast-3-characters-See.below.for.specific.naming.rules, Rule>>
-| Role variables MUST have a prefix of at least 3 characters. See below for specific naming rules.
+| <<Ansible-role-variable-prefix, Rule>>
+| Role variables MUST have a prefix of `r_<role_name>_`
 |===
 
-==== Role with 3 (or more) words in the name
+.Definition
+Ansible variables contained in or passed into a role.
 
-Take the first letter of each of the words.
-
-.3 word example:
-* Role name: made_up_role
-* Prefix: mur
-[source]
-----
-mur_var1: value_one
-----
+.Purpose
+Name variables to avoid name collision between roles or playbooks.
```

Comment:
If we go with @kwoodson's suggestion:
```yaml
- name: A play
  tasks:
  - include_role:
      name: etcd_common
    vars:
      r_etcd_common_etcd_runtime: "{{ openshift.common.etcd_runtime }}"
  - include_role:
      name: etcd_upgrade
    vars:
      r_etcd_upgrade_action: upgrade
      r_etcd_upgrade_mechanism: rpm
      r_etcd_upgrade_version: "3.1.9"
      r_etcd_upgrade_etcd_runtime: "{{ openshift.common.etcd_runtime }}"
      r_etcd_upgrade_service: "{{ r_etcd_common_service }}"
      r_etcd_upgrade_service_file: "{{ r_etcd_common_service_file }}"
      r_etcd_upgrade_etcdctlv2: "{{ r_etcd_common_etcdctlv2 }}"
```

with moving some variables from `etcd_common/defaults/main.yml` and setting them in `etcd_common/tasks/main.yml` via `set_fact` so all "exported" etcd_common role variables are evaluated, my previous comments can be ignored.